// @dcloudio/vite-plugin-uni/dist/config/css.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createCss = void 0;
const path_1 = __importDefault(require("path"));
const fs_extra_1 = __importDefault(require("fs-extra"));
function resolveAdditionalData(inputDir, config) {
    const uniScssFile = path_1.default.resolve(inputDir, 'uni.scss');
    const userAdditionalData = config.css?.preprocessorOptions?.scss?.additionalData || '';
    if (!fs_extra_1.default.existsSync(uniScssFile)) {
        return userAdditionalData;
    }
    return fs_extra_1.default.readFileSync(uniScssFile, 'utf8') + '\n' + userAdditionalData;
}
function createCss(options, config) {
    return {
        preprocessorOptions: {
            scss: {
                charset: false,
                additionalData: resolveAdditionalData(options.inputDir, config),
            },
        },
    };
}
exports.createCss = createCss;
